pkg/installer/installer.go

```diff
@@ -18,100 +18,246 @@ package installer
 
 import (
 	"context"
-	"errors"
 	"fmt"
-	"os"
 	"strconv"
 	"strings"
 
 	"github.com/fatih/color"
 	"github.com/minio/directpv/pkg/k8s"
+	"github.com/minio/directpv/pkg/utils"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/klog/v2"
 )
 
-func trimMinorVersion(minor string) (string, error) {
-	i := strings.IndexFunc(minor, func(r rune) bool { return r < '0' || r > '9' })
-	if i < 0 {
-		return minor, nil
-	}
+var (
+	tick  = color.HiGreenString("✓\n")
+	cross = color.HiRedString("✗\n")
+)
 
-	m := minor[:i]
-	_, err := strconv.Atoi(m)
+func getKubeVersion() (major, minor uint, err error) {
+	versionInfo, err := k8s.DiscoveryClient().ServerVersion()
 	if err != nil {
-		return "", err
+		return 0, 0, err
 	}
 
-	return m, nil
-}
-
-func getInstaller(config *Config) (installer, error) {
-	versionInfo, err := k8s.DiscoveryClient().ServerVersion()
-	if err != nil {
-		return nil, err
+	var u64 uint64
+	if u64, err = strconv.ParseUint(versionInfo.Major, 10, 64); err != nil {
+		return 0, 0, fmt.Errorf("unable to parse major version %v; %v", versionInfo.Major, err)
 	}
+	major = uint(u64)
 
-	minor := versionInfo.Minor
+	minorString := versionInfo.Minor
 	if strings.Contains(versionInfo.GitVersion, "-eks-") {
 		// Do trimming only for EKS.
 		// Refer https://github.com/aws/containers-roadmap/issues/1404
-		minor, err = trimMinorVersion(versionInfo.Minor)
+		i := strings.IndexFunc(minorString, func(r rune) bool { return r < '0' || r > '9' })
+		if i > -1 {
+			minorString = minorString[:i]
+		}
+	}
+	if u64, err = strconv.ParseUint(minorString, 10, 64); err != nil {
+		return 0, 0, fmt.Errorf("unable to parse minor version %v; %v", minor, err)
+	}
+	minor = uint(u64)
+
+	return major, minor, nil
+}
+
+// Install performs DirectPV installation on kubernetes.
+func Install(ctx context.Context, args *Args) error {
+	err := args.validate()
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case args.DryRun:
+		if args.KubeVersion == nil {
+			// default higher version
+			if args.KubeVersion, err = version.ParseSemantic("1.15.0"); err != nil {
```

Comment:
can we chose a higher version incase if kubeVersion is not provided? `1.15.0` is very old and deprecated

Reply:
we can choose a latest version

Reply:
Or can we make `--kube-version` as mandatory for dry-run mode?

Reply:
Done. It was supposed to be `1.25.0`